### DISCUSSION TITLE
Linking.openURL to open Explorer

### DISCUSSION THREAD

**Discussion by WilliamWelsh** [microsoft/react-native-windows#12252] [Q&A] [answered]
Is there a way to use Linking.openURL to open a folder via File Explorer or any file? I tried `file:///${uri}` but it says the URL cannot be opened.

**Comment by chrisglein** [comment, marked as answer]
`file://` isn't called out as on the of the expected protocols from RN core: https://reactnative.dev/docs/0.71/linking#built-in-url-schemes
Internally for RNW we forward that to [LaunchUriAsync](https://learn.microsoft.com/en-us/uwp/api/windows.system.launcher.launchuriasync?view=winrt-22621). Which has more options... but local files aren't one of them:

> You cannot use this method to launch a URI in the local zone. For example, apps cannot use the file:/// protocol to access files on the local computer. Instead, you must use the [Storage APIs](https://learn.microsoft.com/en-us/uwp/api/windows.storage?view=winrt-22621) to access files.

It'd take more work to make RNW's implementation of `Linking` both smarter than RN core and smarter than UWP. But it also seems kind of reasonable. Can you open a feature request on the repo for us to evaluate?

In the meantime you'd need to direct this to your own native handling.
